### ISSUE TITLE
Drawing Tools: new marks cause screen to jump/scroll on Firefox + touchscreens/mobile

### ISSUE THREAD

**Issue by shaunanoordin** [zooniverse/front-end-monorepo#3815]
## UI/UX Bug

Affects: Classifier, Drawing Tools
Noted on: d83bd0cfa9f09734335aa3c45f1a8ec05af78c06
Tested on:
- Browserstack emulated Firefox + Android
- macOS + Firefox 106, touchscreen mode
- ⚠️ NOT yet tested with an actual physical device running Firefox

When creating a new mark on a Drawing Task, the window jumps/scrolls abruptly with the mark at the top edge of the screen.

https://user-images.githubusercontent.com/13952701/197874878-68d16d0e-180f-494a-887b-3c1c1879a1bd.mov

Expectation: view of the screen shouldn't jump/scroll so much that the mark goes off-screen and the Subject image is no longer centred to what the user was previously viewing. On Chrome + Android and iOS + Safari, no such jumping/scrolling occurs. 

### Dev Notes

This may be related to `lib-classifier/src/plugins/drawingTools/components/Mark/Mark.js`:

```
function focusMark() {
  ...
  if (!hasFocus) {
    ...
    window.scrollTo(x, y)  // This bit, maybe?
  }
}
```

EDIT: Maybe not, I took a quick look into this, and if this _was_ the problem, it should also have been affecting Chrome and Firefox-when-not-in-touchscreen mode. It's likely something specifically with Firefox's handling of touch events, it seems.

### Status

The problem exists on PFE as well, so fixing this is technically an enhancement, not a "bring things to PFE parity" fix.

**Comment by eatyourgreens** [comment]
See also https://github.com/zooniverse/Panoptes-Front-End/pull/4122

**Comment by eatyourgreens** [comment]
@shaunanoordin the `focusMark` function is called by marks with subtasks. Does it run for points that don’t have tasks?

This function scrolls a mark into view when it is selected. 
https://github.com/zooniverse/front-end-monorepo/blob/14bebb46b3cd43e4f499c8479485a3ab794ae37c/packages/lib-classifier/src/plugins/drawingTools/components/Mark/Mark.js#L117-L120

**Comment by eatyourgreens** [comment]
I've started the [`mark-effect-hooks`](https://github.com/zooniverse/front-end-monorepo/tree/mark-effect-hooks) branch to look at how focus is managed by the drawing tools.

**Comment by eatyourgreens** [comment]
I've opened #3817 to look into this. All new drawing tool marks are automatically focused when you create them. On desktop, this allows you to immediately control them from the keyboard: zoom in/out, pan with the cursor keys, open sub tasks with Enter/Space or delete the new mark with Backspace.

I was going to say that none of those keyboard interactions are available on iOS or Android, but you could be using a tablet with a bluetooth keyboard. 🤔

**Comment by eatyourgreens** [comment]
I also found that new marks aren't always being focused when you create them, so I've fixed that in #3817.

**Comment by goplayoutside3** [comment]
I tried to replicate this in Firefox 121 touch screen emulation on [Elephant ID](https://www.zooniverse.org/projects/aeuk/elephant-id)'s drawing tools workflow, but no longer see any jump/scroll. Please feel free to re-open if reports come in from Firefox on a mobile device.
